src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp

```diff
@@ -104,66 +120,109 @@ ExitCode PlatformInconsistencyCheckerWorker::checkTree(std::shared_ptr<Node> rem
+            if (!_syncPal->syncDb()->dbId(ReplicaSide::Local, *localNode->id(), dbId, found)) {
+                LOGW_WARN(_logger,
+                          L"Failed to retreive dbId for local node: " << Utility::formatSyncPath(absoluteLocalPath).c_str());
+            }
+            if (found && !_syncPal->syncDb()->deleteNode(dbId, found)) {
+                // Remove node (and childs by cascade) from DB if it exists (else ignore as it is already not in DB)
+                LOGW_WARN(_logger, L"Failed to delete node from DB: " << Utility::formatSyncPath(absoluteLocalPath).c_str());
+            }
 
-        if (localNode->id().has_value()) nodeIDs.localId = *localNode->id();
+            if (!_syncPal->vfsFileStatusChanged(absoluteLocalPath, node->side() == ReplicaSide::Remote ? SyncFileStatus::Conflict
+                                                                                                       : SyncFileStatus::Error)) {
+                LOGW_SYNCPAL_WARN(_logger, L"Error in SyncPal::vfsFileStatusChanged: "
+                                                   << Utility::formatSyncPath(absoluteLocalPath).c_str());
+            }
+        }
     }
 
-    _syncPal->blacklistTemporarily(remoteNode->id().value(), relativePath, remoteNode->side());
-    Error error(_syncPal->syncDbId(), "", remoteNode->id().value(), remoteNode->type(), relativePath, ConflictType::None,
+    if (node->side() == ReplicaSide::Remote) {
+        _syncPal->blacklistTemporarily(remoteNode->id().value(), remoteNode->getPath(), remoteNode->side());
+    }
+
+    Error error(_syncPal->syncDbId(), "", node->id().value(), node->type(), node->getPath(), ConflictType::None,
                 inconsistencyType);
     _syncPal->addError(error);
 
-    std::wstring causeStr;
-    switch (inconsistencyType) {
-        case InconsistencyType::Case:
-            causeStr = L"of a name clash";
-            break;
-        case InconsistencyType::ForbiddenChar:
-            causeStr = L"of a forbidden character";
-            break;
-        case InconsistencyType::ReservedName:
-            causeStr = L"the name is reserved on this OS";
-            break;
-        case InconsistencyType::NameLength:
-            causeStr = L"the name is too long";
-            break;
-        default:
-            break;
-    }
-    LOGW_SYNCPAL_INFO(_logger, L"Blacklisting remote item with " << Utility::formatSyncPath(relativePath).c_str() << L" because "
-                                                                 << causeStr.c_str() << L".");
-
-    nodeIDs.remoteId = *remoteNode->id();
+    LOGW_SYNCPAL_INFO(_logger, L"Blacklisting " << node->side() << L" item with "
+                                                << Utility::formatSyncPath(node->getPath()).c_str() << L" because "
+                                                << inconsistencyType << L".");
+
+    nodeIDs.remoteId = (remoteNode && remoteNode->id().has_value()) ? *remoteNode->id() : NodeId();
```

Comment:
A small helper function (might be a lambda) would avoid code duplication.

Reply:
Done here: [4f28977](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/4f28977c75d813569308efb36f9e5f22fea5f921)